superset/security/api.py

```diff
@@ -0,0 +1,62 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+
+from flask import Response
+from flask_appbuilder import expose
+from flask_appbuilder.api import BaseApi, safe
+from flask_appbuilder.security.decorators import permission_name, protect
+from flask_wtf.csrf import generate_csrf
+
+from superset.extensions import event_logger
+
+logger = logging.getLogger(__name__)
+
+
+class SecurityRestApi(BaseApi):
+    resource_name = "security"
+    allow_browser_login = True
+    openapi_spec_tag = "Security"
+
+    @expose("/csrf_token/", methods=["GET"])
+    @event_logger.log_this
+    @protect()
+    @safe
+    @permission_name("read")
```

Comment:
Do we need to add this new permission in FAB to existing roles?

Reply:
They get automatically added on `superset init` to all roles

Reply:
You could add a test to make sure a CSRF token isn't issued when the user is not logged in - that would be my recommended way forward there. Call me paranoid, it's just that this endpoint is part of the security story for Superset.

Reply:
paranoya is wellcome on security related matters

Reply:
@willbarrett Actually this led to a small refactor on tests. IMO having a public role like `Gamma` by default on all tests was dangerous, and by disabling it surfaced a bunch of tests that were not authenticated or handling anonymous users on a not 100% correct way. 
Now setting the public role like `Gamma` is a fixture, so it's easy to use when needed